File: 10110/main.c
#include <stdio.h>
#include <math.h>

int main(int argc, char* argv[])
{
   int n;
   double s;
   char yes;

   while (scanf("%d", &n) == 1 && n != 0) {
      s = sqrt((double) n);
      yes = (((int) s * (int) s) == n);
      printf("%s\n", (yes == 1) ? "yes" : "no");
   }
   return 0;
}

File: 10110/Main.java
import java.io.*;

public class Main {
   public static void main(String args[]) throws IOException {
      String line;
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      while ((line = in.readLine()) != null) {
         if (line.charAt(0) == '0') break;
         long n = Long.parseLong(line);
         double s = Math.sqrt(n);
         if (((long) s * (long) s) == n) {
            System.out.println("yes");
         } else {
            System.out.println("no");
         }
      }
   }
}
